maps-ktx/src/main/java/com/google/maps/android/ktx/GoogleMap.kt

```diff
@@ -390,7 +383,7 @@ public fun GoogleMap.mapLongClickEvents(): Flow<LatLng> =
 public fun GoogleMap.markerClickEvents(): Flow<Marker> =
     callbackFlow {
         setOnMarkerClickListener {
-            offerCatching(it)
+            trySend(it).isSuccess
```

Comment:
Just curious - why do only some of these use `.isSuccess`?

Reply:
This is because some callback methods return a boolean (like in this case [onMarkerClick](https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap.OnMarkerClickListener#public-abstract-boolean-onmarkerclick-marker-marker)). `offerCatching` returned boolean which was being swallowed in void callback methods. The implementation here of returning `isSuccess` should be equivalent to calling `offerCatching`.

Looking at this method a bit more though, I think the return value should be customizable by the caller. As it is currently implemented, if the marker is clicked and the channel is able to send the marker object, the default behavior of moving the camera to center on the marker and displaying an info window will always occur. Will file an issue for this.

Reply:
Issue filed here https://github.com/googlemaps/android-maps-ktx/issues/163

Reply:
Ah, ok - thanks. That's what I initially thought but I was confusing myself by looking at the wrong methods - in this case, [setOnMarkerClickListener](https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap#setOnMarkerClickListener(com.google.android.gms.maps.GoogleMap.OnMarkerClickListener)), which returns `void` instead of onMarkerClick, which returns `boolean`.

>I think the return value should be customizable by the caller. 

I agree, that makes sense 👍